### PR TITLE
feat: SVPI-61 - vault deploy, init and configure

### PR DESCRIPTION
**Pull request by sparkoo**
### What does this PR do?
vault deployment and configuration on minikube or openshift

 - minikube `make deploy_minikube`
 - openshift `make deploy`
 - script to init vault 
   - initialize vault and stores unseal keys into `spi-vault-keys` secret
   - setup kubernetes auth
   - setup audit log
   - setup `spi` kv secret engine
 - automatically unseal vault at pod startup
 - script for root access
 - generate new deployment yamls with `./hack/vault-generate-template.sh`
 - updates README

I've tested on minikube, openshift deployed with `make deploy`, openshift with infra-deployments

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-61

### How to test this PR?

1. have `kubectl` configured for desired cluster, openshift or minikube
2. run `make deploy_minikube` or `make deploy` for openshift
3. Vault should be initialized and configured. You can go to Vault shell, run `vault status`. Should be initialized and unsealsed
4. To gain root access on vault pod, run `sh /vault/userconfig/scripts/root.sh`. This generates root token and login with it.

### CODE REVIEW

File: hack/vault-init.sh [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+init
+secret
+unseal
+ensureRootToken
+login
+audit
+k8sAuth
+spiSecretEngine
+restart
```
Comment (skabashnyuk) [inline hack/vault-init.sh:157]: do we need to keep logged in state after restart?

File: config/vault/openshift/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: VAULT_CLUSTER_ADDR
+              value: "https://$(HOSTNAME).vault-internal:8201"
+            - name: HOME
+              value: "/home/vault"
+            - name: VAULT_LOG_LEVEL
+              value: "trace"
+
```
Comment (skabashnyuk) [inline config/vault/openshift/deployment.yaml:234]: can you remove unnecessary spaces?

File: hack/vault-init.sh (hunk 1 above)
Comment (sparkoo) [reply hack/vault-init.sh:157]: it won't login after restart. Or do you see it logged in?

File: config/vault/openshift/deployment.yaml (hunk 2 above)
Comment (sparkoo) [reply config/vault/openshift/deployment.yaml:234]: these are generated by helm see `vault-generate-template.sh` script. I would not modify them manually. I can somehow update the script to remove empty lines, but I'm not sure it's worth it.

File: hack/vault-init.sh (hunk 1 above)
Comment (skabashnyuk) [reply hack/vault-init.sh:157]: I've missed the point where the code made logout. is there the one?

File: config/vault/openshift/deployment.yaml (hunk 2 above)
Comment (skabashnyuk) [reply config/vault/openshift/deployment.yaml:234]: ok

File: hack/vault-init.sh (hunk 1 above)
Comment (skabashnyuk) [reply hack/vault-init.sh:157]: ok

**Review by skabashnyuk — APPROVED** [review]
Looks good to me. what would be our next steps in the path to deploy and use vault on staging?

**Pull request merged**